lib/jekyll-admin/server/collection.rb

```diff
@@ -29,7 +29,9 @@ class Server < Sinatra::Base
         end
 
         write_file(write_path, document_body)
-        updated_document = collection.docs.find { |d| d.path == write_path }
+        updated_document = Jekyll::Document.new(write_path, {
+          :collection => collection, :site => site,
+        }).tap(&:read)
         render_404 if updated_document.nil?
```

Comment:
We won't be able to do this, because `ensure_document` calls `document`, which uses the buggy way of reading:

```ruby
 80       def document
 81         collection.docs.find { |d| d.path == document_path }
 82       end
```